Android debug build fails

<details>

<summary>x86</summary>

```
$ ./kodev build android-x86 --debug --verbose
make TARGET= KODEBUG=1 VERBOSE=1 android-ndk android-sdk
make -C base/toolchain /home/frans/src/kobo/koreader/base/toolchain/android-ndk-r23c
make[1]: Entering directory '/home/frans/src/kobo/koreader/base/toolchain'
make[1]: '/home/frans/src/kobo/koreader/base/toolchain/android-ndk-r23c' is up to date.
make[1]: Leaving directory '/home/frans/src/kobo/koreader/base/toolchain'
make -C base/toolchain /home/frans/src/kobo/koreader/base/toolchain/android-sdk-linux
make[1]: Entering directory '/home/frans/src/kobo/koreader/base/toolchain'
make[1]: '/home/frans/src/kobo/koreader/base/toolchain/android-sdk-linux' is up to date.
make[1]: Leaving directory '/home/frans/src/kobo/koreader/base/toolchain'
make ANDROID_ARCH=x86 TARGET=android KODEBUG=1 VERBOSE=1
cd base/build/i686-unknown-linux-android18-debug/cmake && ninja -v -j16 -l16 all
▸  0% | cd /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/build && : "" && set -- android-luajit-launcher "Building 'android-luajit-launcher'" 0 /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/stamp/build && . /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koenv.sh && run ninja && /usr/bin/cmake -E cmake_transform_depfile Ninja gccdepfile /home/frans/src/kobo/koreader/base/cmake /home/frans/src/kobo/koreader/base/thirdparty/android-luajit-launcher /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/android-luajit-launcher /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/build.d /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/CMakeFiles/d/c81c39cfad6ee36d4daea53e21221d6b95580a27791234cd4b4e62fde46ea6eb.d
+ ninja
ninja: no work to do.
▸ 16% | cd /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/build && : "" && set -- crengine "Building 'crengine'" 0 /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/stamp/build && . /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koenv.sh && run ninja
+ ninja
ninja: no work to do.
▸ 50% | cd /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/build && : "" && set -- crengine "Installing 'crengine'" 0 /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/stamp/install /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/log/install && . /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koenv.sh
▸ 66% | cd /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/build && : "" && set -- android-luajit-launcher "Installing 'android-luajit-launcher'" 0 /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/stamp/install /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/log/install && . /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koenv.sh && run /usr/bin/cmake --install . && run /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs && run llvm-objcopy --only-keep-debug /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libioctl.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libioctl.so.dbg && run llvm-objcopy --add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libioctl.so.dbg --strip-unneeded /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libioctl.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libioctl.so && run chmod +x /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libioctl.so && run /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs && run llvm-objcopy --only-keep-debug /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libluajit-launcher.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libluajit-launcher.so.dbg && run llvm-objcopy --add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libluajit-launcher.so.dbg --strip-unneeded /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libluajit-launcher.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libluajit-launcher.so && run chmod +x /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libluajit-launcher.so
+ /usr/bin/cmake --install .
-- Install configuration: "Debug"
-- Up-to-date: /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/lib7z.a
-- Up-to-date: /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libioctl.so
-- Up-to-date: /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libluajit-launcher.so
+ /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs
+ llvm-objcopy --only-keep-debug /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libioctl.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libioctl.so.dbg
+ llvm-objcopy '--add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libioctl.so.dbg' --strip-unneeded /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libioctl.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libioctl.so
+ chmod +x /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libioctl.so
+ /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs
+ llvm-objcopy --only-keep-debug /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libluajit-launcher.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libluajit-launcher.so.dbg
+ llvm-objcopy '--add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libluajit-launcher.so.dbg' --strip-unneeded /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libluajit-launcher.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libluajit-launcher.so
+ chmod +x /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libluajit-launcher.so
▸ 66% | cd /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koreader/build && : "" && set -- koreader "Building 'koreader'" 0 /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koreader/stamp/build && . /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koenv.sh && run ninja
+ ninja
      | [koreader 100%] : && /home/frans/src/kobo/koreader/base/toolchain/android-ndk-r23c/toolchains/llvm/prebuilt/linux-x86_64/bin/i686-linux-android18-clang++ -fPIC -Og -g -pipe -fno-omit-frame-pointer -fdata-sections -ffunction-sections -ffile-prefix-map=/home/frans/src/kobo/koreader/base/= -Wno-ignored-optimization-argument -Wno-unknown-warning-option -Wno-array-bounds -Wno-unused-command-line-argument -Wno-self-assign -Wno-parentheses-equality -Wno-macro-redefined -std=gnu++17 -fPIC -g  -Wl,-O1,--as-needed,--gc-sections -static-libstdc++ -no-canonical-prefixes -Wl,--build-id=none   -Wl,-Bsymbolic,--start-group -Wl,@koreader-monolibtic.link_args -shared -Wl,-soname,libkoreader-monolibtic.so -o libkoreader-monolibtic.so CMakeFiles/blitbuffer.dir/blitbuffer.c.o CMakeFiles/koreader-cre.dir/cre.cpp.o CMakeFiles/koreader-djvu.dir/djvu.c.o CMakeFiles/koreader-nnsvg.dir/nnsvg.c.o CMakeFiles/koreader-xtext.dir/xtext.cpp.o CMakeFiles/wrap-mupdf.dir/wrap-mupdf.c.o CMakeFiles/koreader-monolibtic.dir/monolibtic.cpp.o  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libzstd.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/build/libcrengine.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/build/crengine/thirdparty/antiword/libantiword.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/build/crengine/thirdparty/chmlib/libchmlib.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libpng16.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/liblunasvg.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libmd4c-html.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libmd4c.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libutf8proc.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libdjvulibre.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libk2pdfopt.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libcpu_features.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libndk_compat.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libtesseract.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libluajit.so  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libfribidi.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libunibreak.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libjpeg.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libfreetype.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libharfbuzz.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libwebp.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libwebpdemux.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libz.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libmupdf.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libmupdf-third.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libaes.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/liblzma.a  -lm  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/lib7z.a  -landroid  -llog  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libarchive.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libczmq.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libgif.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libkoreader-lfs.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libleptonica.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libturbojpeg.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libcrypto.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libssl.a  -pthread  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libzmq.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/liblodepng.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/lpeg.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/rapidjson.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/ssl.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libluasocket.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/mcore.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/score.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libsqlite3.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libtffi_wrap.a && cd /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koreader/build && /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs && llvm-objcopy --only-keep-debug libkoreader-monolibtic.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so.dbg && llvm-objcopy --add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so.dbg --strip-unneeded libkoreader-monolibtic.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so && chmod +x /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so
FAILED: libkoreader-monolibtic.so 
: && /home/frans/src/kobo/koreader/base/toolchain/android-ndk-r23c/toolchains/llvm/prebuilt/linux-x86_64/bin/i686-linux-android18-clang++ -fPIC -Og -g -pipe -fno-omit-frame-pointer -fdata-sections -ffunction-sections -ffile-prefix-map=/home/frans/src/kobo/koreader/base/= -Wno-ignored-optimization-argument -Wno-unknown-warning-option -Wno-array-bounds -Wno-unused-command-line-argument -Wno-self-assign -Wno-parentheses-equality -Wno-macro-redefined -std=gnu++17 -fPIC -g  -Wl,-O1,--as-needed,--gc-sections -static-libstdc++ -no-canonical-prefixes -Wl,--build-id=none   -Wl,-Bsymbolic,--start-group -Wl,@koreader-monolibtic.link_args -shared -Wl,-soname,libkoreader-monolibtic.so -o libkoreader-monolibtic.so CMakeFiles/blitbuffer.dir/blitbuffer.c.o CMakeFiles/koreader-cre.dir/cre.cpp.o CMakeFiles/koreader-djvu.dir/djvu.c.o CMakeFiles/koreader-nnsvg.dir/nnsvg.c.o CMakeFiles/koreader-xtext.dir/xtext.cpp.o CMakeFiles/wrap-mupdf.dir/wrap-mupdf.c.o CMakeFiles/koreader-monolibtic.dir/monolibtic.cpp.o  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libzstd.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/build/libcrengine.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/build/crengine/thirdparty/antiword/libantiword.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/thirdparty/crengine/build/crengine/thirdparty/chmlib/libchmlib.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libpng16.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/liblunasvg.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libmd4c-html.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libmd4c.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libutf8proc.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libdjvulibre.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libk2pdfopt.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libcpu_features.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libndk_compat.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libtesseract.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libluajit.so  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libfribidi.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libunibreak.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libjpeg.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libfreetype.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libharfbuzz.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libwebp.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libwebpdemux.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libz.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libmupdf.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libmupdf-third.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libaes.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/liblzma.a  -lm  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/lib7z.a  -landroid  -llog  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libarchive.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libczmq.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libgif.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libkoreader-lfs.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libleptonica.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libturbojpeg.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libcrypto.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libssl.a  -pthread  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libzmq.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/liblodepng.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/lpeg.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/rapidjson.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/ssl.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libluasocket.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/mcore.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/score.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libsqlite3.a  /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libtffi_wrap.a && cd /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koreader/build && /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs && llvm-objcopy --only-keep-debug libkoreader-monolibtic.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so.dbg && llvm-objcopy --add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so.dbg --strip-unneeded libkoreader-monolibtic.so /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so && chmod +x /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so
ld: error: undefined hidden symbol: hb_buffer_t::message_impl(hb_font_t*, char const*, char*)
>>> referenced by hb-buffer-verify.cc:51 (../source/src/hb-buffer-verify.cc:51)
>>>               hb-buffer-verify.cc.o:(buffer_verify_error(hb_buffer_t*, hb_font_t*, char const*, ...)) in archive /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/staging/lib/libharfbuzz.a
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
FAILED: koreader/CMakeFiles/koreader-build koreader/stamp/build /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koreader/CMakeFiles/koreader-build /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koreader/stamp/build 
cd /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koreader/build && : "" && set -- koreader "Building 'koreader'" 0 /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koreader/stamp/build && . /home/frans/src/kobo/koreader/base/build/i686-unknown-linux-android18-debug/cmake/koenv.sh && run ninja
ninja: build stopped: subcommand failed.
make: *** [base/Makefile:86: base-all] Error 1
```

</details>


<details>

<summary>arm</summary>


```
$ ./kodev build android-arm --debug --verbose
make TARGET= KODEBUG=1 VERBOSE=1 android-ndk android-sdk
make -C base/toolchain /home/frans/src/kobo/koreader/base/toolchain/android-ndk-r23c
make[1]: Entering directory '/home/frans/src/kobo/koreader/base/toolchain'
make[1]: '/home/frans/src/kobo/koreader/base/toolchain/android-ndk-r23c' is up to date.
make[1]: Leaving directory '/home/frans/src/kobo/koreader/base/toolchain'
make -C base/toolchain /home/frans/src/kobo/koreader/base/toolchain/android-sdk-linux
make[1]: Entering directory '/home/frans/src/kobo/koreader/base/toolchain'
make[1]: '/home/frans/src/kobo/koreader/base/toolchain/android-sdk-linux' is up to date.
make[1]: Leaving directory '/home/frans/src/kobo/koreader/base/toolchain'
make ANDROID_ARCH=arm TARGET=android KODEBUG=1 VERBOSE=1
cd base/build/armv7a-unknown-linux-android18-debug/cmake && ninja -v -j16 -l16 all
▸  0% | cd /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/build && : "" && set -- android-luajit-launcher "Building 'android-luajit-launcher'" 0 /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/stamp/build && . /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koenv.sh && run ninja && /usr/bin/cmake -E cmake_transform_depfile Ninja gccdepfile /home/frans/src/kobo/koreader/base/cmake /home/frans/src/kobo/koreader/base/thirdparty/android-luajit-launcher /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/android-luajit-launcher /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/build.d /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/CMakeFiles/d/547bb8a11b179b8dfc509cd0a87c7446df28128eaa8b9ac62236d87f3ce786a8.d
+ ninja
ninja: no work to do.
▸ 16% | cd /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/build && : "" && set -- crengine "Building 'crengine'" 0 /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/stamp/build && . /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koenv.sh && run ninja
+ ninja
ninja: no work to do.
▸ 50% | cd /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/build && : "" && set -- crengine "Installing 'crengine'" 0 /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/stamp/install /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/log/install && . /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koenv.sh
▸ 66% | cd /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/build && : "" && set -- android-luajit-launcher "Installing 'android-luajit-launcher'" 0 /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/stamp/install /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/android-luajit-launcher/log/install && . /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koenv.sh && run /usr/bin/cmake --install . && run /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs && run llvm-objcopy --only-keep-debug /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libioctl.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libioctl.so.dbg && run llvm-objcopy --add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libioctl.so.dbg --strip-unneeded /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libioctl.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libioctl.so && run chmod +x /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libioctl.so && run /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs && run llvm-objcopy --only-keep-debug /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libluajit-launcher.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libluajit-launcher.so.dbg && run llvm-objcopy --add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libluajit-launcher.so.dbg --strip-unneeded /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libluajit-launcher.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libluajit-launcher.so && run chmod +x /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libluajit-launcher.so
+ /usr/bin/cmake --install .
-- Install configuration: "Debug"
-- Up-to-date: /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/lib7z.a
-- Up-to-date: /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libioctl.so
-- Up-to-date: /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libluajit-launcher.so
+ /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs
+ llvm-objcopy --only-keep-debug /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libioctl.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libioctl.so.dbg
+ llvm-objcopy '--add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libioctl.so.dbg' --strip-unneeded /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libioctl.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libioctl.so
+ chmod +x /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libioctl.so
+ /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs
+ llvm-objcopy --only-keep-debug /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libluajit-launcher.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libluajit-launcher.so.dbg
+ llvm-objcopy '--add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libluajit-launcher.so.dbg' --strip-unneeded /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libluajit-launcher.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libluajit-launcher.so
+ chmod +x /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libluajit-launcher.so
▸ 66% | cd /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koreader/build && : "" && set -- koreader "Building 'koreader'" 0 /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koreader/stamp/build && . /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koenv.sh && run ninja
+ ninja
      | [koreader 100%] : && /home/frans/src/kobo/koreader/base/toolchain/android-ndk-r23c/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi18-clang++ -fPIC -Og -g -pipe -fno-omit-frame-pointer -fdata-sections -ffunction-sections -ffile-prefix-map=/home/frans/src/kobo/koreader/base/= -Wno-ignored-optimization-argument -Wno-unknown-warning-option -Wno-array-bounds -Wno-unused-command-line-argument -Wno-self-assign -Wno-parentheses-equality -Wno-macro-redefined -std=gnu++17 -march=armv7-a -mfpu=vfpv3-d16 -mthumb -funwind-tables -fstack-protector -no-canonical-prefixes -mfloat-abi=softfp -fPIC -g  -Wl,-O1,--as-needed,--gc-sections -static-libstdc++ -no-canonical-prefixes -Wl,--fix-cortex-a8 -march=armv7-a -Wl,--build-id=none   -Wl,-Bsymbolic,--start-group -Wl,@koreader-monolibtic.link_args -shared -Wl,-soname,libkoreader-monolibtic.so -o libkoreader-monolibtic.so CMakeFiles/blitbuffer.dir/blitbuffer.c.o CMakeFiles/koreader-cre.dir/cre.cpp.o CMakeFiles/koreader-djvu.dir/djvu.c.o CMakeFiles/koreader-nnsvg.dir/nnsvg.c.o CMakeFiles/koreader-xtext.dir/xtext.cpp.o CMakeFiles/wrap-mupdf.dir/wrap-mupdf.c.o CMakeFiles/koreader-monolibtic.dir/monolibtic.cpp.o -L/home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libzstd.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/build/libcrengine.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/build/crengine/thirdparty/antiword/libantiword.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/build/crengine/thirdparty/chmlib/libchmlib.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libpng16.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/liblunasvg.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libmd4c-html.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libmd4c.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libutf8proc.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libdjvulibre.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libk2pdfopt.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libcpu_features.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libndk_compat.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libtesseract.a  -lluajit  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libfribidi.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libunibreak.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libjpeg.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libfreetype.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libharfbuzz.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libwebp.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libwebpdemux.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libz.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libmupdf.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libmupdf-third.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libaes.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/liblzma.a  -lm  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/lib7z.a  -landroid  -llog  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libarchive.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libczmq.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libgif.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libkoreader-lfs.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libleptonica.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libturbojpeg.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libcrypto.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libssl.a  -pthread  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libzmq.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/liblodepng.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/lpeg.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/rapidjson.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/ssl.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libluasocket.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/mcore.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/score.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libsqlite3.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libtffi_wrap.a && cd /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koreader/build && /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs && llvm-objcopy --only-keep-debug libkoreader-monolibtic.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so.dbg && llvm-objcopy --add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so.dbg --strip-unneeded libkoreader-monolibtic.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so && chmod +x /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so
FAILED: libkoreader-monolibtic.so 
: && /home/frans/src/kobo/koreader/base/toolchain/android-ndk-r23c/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi18-clang++ -fPIC -Og -g -pipe -fno-omit-frame-pointer -fdata-sections -ffunction-sections -ffile-prefix-map=/home/frans/src/kobo/koreader/base/= -Wno-ignored-optimization-argument -Wno-unknown-warning-option -Wno-array-bounds -Wno-unused-command-line-argument -Wno-self-assign -Wno-parentheses-equality -Wno-macro-redefined -std=gnu++17 -march=armv7-a -mfpu=vfpv3-d16 -mthumb -funwind-tables -fstack-protector -no-canonical-prefixes -mfloat-abi=softfp -fPIC -g  -Wl,-O1,--as-needed,--gc-sections -static-libstdc++ -no-canonical-prefixes -Wl,--fix-cortex-a8 -march=armv7-a -Wl,--build-id=none   -Wl,-Bsymbolic,--start-group -Wl,@koreader-monolibtic.link_args -shared -Wl,-soname,libkoreader-monolibtic.so -o libkoreader-monolibtic.so CMakeFiles/blitbuffer.dir/blitbuffer.c.o CMakeFiles/koreader-cre.dir/cre.cpp.o CMakeFiles/koreader-djvu.dir/djvu.c.o CMakeFiles/koreader-nnsvg.dir/nnsvg.c.o CMakeFiles/koreader-xtext.dir/xtext.cpp.o CMakeFiles/wrap-mupdf.dir/wrap-mupdf.c.o CMakeFiles/koreader-monolibtic.dir/monolibtic.cpp.o -L/home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libzstd.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/build/libcrengine.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/build/crengine/thirdparty/antiword/libantiword.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/thirdparty/crengine/build/crengine/thirdparty/chmlib/libchmlib.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libpng16.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/liblunasvg.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libmd4c-html.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libmd4c.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libutf8proc.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libdjvulibre.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libk2pdfopt.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libcpu_features.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libndk_compat.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libtesseract.a  -lluajit  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libfribidi.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libunibreak.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libjpeg.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libfreetype.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libharfbuzz.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libwebp.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libwebpdemux.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libz.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libmupdf.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libmupdf-third.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libaes.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/liblzma.a  -lm  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/lib7z.a  -landroid  -llog  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libarchive.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libczmq.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libgif.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libkoreader-lfs.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libleptonica.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libturbojpeg.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libcrypto.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libssl.a  -pthread  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libzmq.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/liblodepng.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/lpeg.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/rapidjson.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/ssl.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libluasocket.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/mcore.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/score.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libsqlite3.a  /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libtffi_wrap.a && cd /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koreader/build && /usr/bin/cmake -E make_directory /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs && llvm-objcopy --only-keep-debug libkoreader-monolibtic.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so.dbg && llvm-objcopy --add-gnu-debuglink=/home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so.dbg --strip-unneeded libkoreader-monolibtic.so /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so && chmod +x /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/libs/libkoreader-monolibtic.so
ld: error: undefined hidden symbol: hb_buffer_t::message_impl(hb_font_t*, char const*, std::__va_list)
>>> referenced by hb-buffer-verify.cc:51 (../source/src/hb-buffer-verify.cc:51)
>>>               hb-buffer-verify.cc.o:(buffer_verify_error(hb_buffer_t*, hb_font_t*, char const*, ...)) in archive /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/staging/lib/libharfbuzz.a
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
FAILED: koreader/CMakeFiles/koreader-build koreader/stamp/build /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koreader/CMakeFiles/koreader-build /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koreader/stamp/build 
cd /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koreader/build && : "" && set -- koreader "Building 'koreader'" 0 /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koreader/stamp/build && . /home/frans/src/kobo/koreader/base/build/armv7a-unknown-linux-android18-debug/cmake/koenv.sh && run ninja
ninja: build stopped: subcommand failed.
make: *** [base/Makefile:86: base-all] Error 1
```

</details>

--- a/thirdparty/harfbuzz/android.patch
+++ b/thirdparty/harfbuzz/android.patch
@@ -1,3 +1,21 @@
+diff --git a/src/hb-buffer-verify.cc b/src/hb-buffer-verify.cc
+index e1ccd5b94..e62b8fb9f 100644
+--- a/src/hb-buffer-verify.cc
++++ b/src/hb-buffer-verify.cc
+@@ -46,11 +46,13 @@ buffer_verify_error (hb_buffer_t *buffer,
+ {
+   va_list ap;
+   va_start (ap, fmt);
++#ifndef HB_NO_BUFFER_MESSAGE
+   if (buffer->messaging ())
+   {
+     buffer->message_impl (font, fmt, ap);
+   }
+   else
++#endif
+   {
+     fprintf (stderr, "harfbuzz ");
+     vfprintf (stderr, fmt, ap);
 diff --git a/src/hb.hh b/src/hb.hh
 index ba18e36b9..aabbb6e97 100644
 --- a/src/hb.hh

